### ISSUE TITLE
inexistent svg notes reported by anno_start

### ISSUE THREAD

**Issue by spheras** [moinejf/abc2svg#20]
Hello!,

it seems I've detected a bug when rendering SVG images from ABC, only for certains dimensions.

You can see the effect at this screenshot from the software I'm developing using abc2svg library (https://spheras.github.io/backingtrainer/).  At this screenshot I use the last version of abc2svg 1.13.7 (currently, the official released version use an old one -1.12.6- which doesn't seems to have the bug)
It seems that the anno_start function, in some cases where one measure need to be split/wrap in two lines like in the screenshot, it reports an inexistent note figure.
Highlighted in blue is the box that is reported.

![screenshot](https://user-images.githubusercontent.com/3862933/29682819-abc3473e-890c-11e7-9a81-8d0fea243874.jpg)
​
The following is an example of ABC where you can see the problem. The anno_start function report 2 inexistent notes. The first one is the number 317 figure reported, the second the 551.

`
X:1
T:J. S. Bach (BWV 244 No. 49)
T:Aria: "Aus Liebe will meinx Heiland sterben"
T:Interpretation for Flute & Harp by Mike Magatagan 2017
Z:Aria: "Aus Liebe will mein Heiland sterben" (BWV 232 No. 49) by J.S. Bach Arranged for Flute & Harp by Mike Magatagan
Z:Mike Magatagan (magataganm@cox.net or Mike Magatagan on http://www.MuseScore.com)
%%scale 1
%%pagewidth 11cm
%%leftmargin 0cm
%%rightmargin 0cm
L:1/16
Q:1/4=48
M:3/4
I:linebreak $
K:C
V:1 treble nm="Flute" snm="F"
%%MIDI program 73
%%MIDI control 7 74
%%MIDI control 10 1968.5
V:1
"^Aria"!mf! A4 | (e6- e/f/e/d/){d} (e2>c'2) |{e} d8 b4- | b(c'ba) (ba^g^f edcB) |$ %4
 (BdcB){B} A4 z (Ace) | (fefa c'bag) fef2- | f(^deg bagf) e=de2- |$ e(^cdf agfe) dcd2- | %8
 d(fed b^gfe) dB^c2- | c(B^ce ag^fe) ^dB=c2- |$ c(Bc^d ^f^gab) c'ab2 | b4- b(fed eBc^G) | %12
 (A/c/^d/^f/a2- a)^g/f/ec'- c'b/a/ga |$ (a3g/f/) (eaed) dbdc | (cBA^G) (ABcd) (efeg) | f12- | %16
 f(c'ba ba^g^f) edb2- |$ bde^G (Aceg) (fb)(fe) | (dcBc){B} A4 z4 | z2 b2- b(agf) ede2- | %20
 e(^cdf agfe) dcd2- |$ d(fed) (b^gfe) dB^c2- | c(B^ce) (ag^fe) ^dB=c2- | c(Bc^d ^f^gab) c'ab2 |$ %24
 b4 z4 z4 | z12 | z12 | z4 z4 g4- |$ g(fed) e(Bce) (abc'a) |{g} f8 d'4- | d'(c'ba gfe)f{f} d'2>f2 | %31
{f} e12- |$ e_ede (ab)(bc') c'(eed |{c} B4- B)_aag g^ffg | %34
 (g/c'/b/a/)(g/c'/)c (c/d/e2)d (Td3c/d/) | c4- cdcB (ce/f/g)e |$ c4- c(dcB) (ce/f/g)e | %37
 (_b3a/g/) f4 (e2>f2) | f4- f(gfe) (fa/^a/)(c'=a) | f4- f(gfe) (fa/_b/)(c'a) |$ %40
 (_e'3d'/c'/) _bg a4 _a2- | a^fg2- g^a(d'c' a)(=a/g/)(fg) | (c^Ac^f) (agf_e dcA=A) | %43
 (_BABd) (ed^cd) (ga)_b2- |$ b^ga2 z4 z4 | a6 (a/_b/a/g/) (a2>d'2) | g12- | g(agf ed^c_B AGFE) |$ %48
 (GA^A^c def^g) (ag)(ba) | a12- | a2^g2 (g2a2) b4- | b2a^g (abc'd') (c'bag) |$ (a4- agfe) (d^cd2- | %53
 d4) z4 z4 | z (fed) (b^gfe dB)^c2- | c(B^ce) (ag^fe) ^dB=c2- |$ c(Bc^d ^f^gab c'a)b2- | b4 z4 z4 | %58
 z12 | z12 | z12 | z4 z4 A4 | (e6- e/f/e/d/){d} (e2>c'2) |{e} d8 b4- |$ b(c'ba) (ba^g^f edcB) | %65
 (BdcB){B} A4 z (Ace) | (fefa c'bag) fef2- | f(^deg bagf) e=de2- |$ e(^cdf agfe) dcd2- | %69
 d(fed) (b^gfe dB)^c2- | c(B^ce ag^fe) ^dB=c2- |$ c(Bc^d ^f^gab c'a)b2 | %72
 b4- b(fed) e[Q:1/4=64.00]"^rit."(Bc^G) | (A/c/^d/^f/a2- a)(^g/f/e)c'- c'(b/a/)(ga) | %74
 !fermata!a12 |] %75`

The user data I put to the library when detecting this problems is:
   `         'imagesize': 'width=320',
            'page_format': true`
`
Is there a way to force the library not to split a measure in two lines? This way I could avoid the problem.
Now I'm detecting, and ignoring, those cases when the box of the reported note is overlapping the clef, but is not working in all the cases.

By the way, thank you for this great library! It is amazing!

**Comment by spheras** [comment]
I have been looking into the code a bit and I discovered that the reported note to anno_start has the invis property true, and it is ignored after when drawing.

Maybe the solution could be ignoring this before calling the anno_start function? Not sure if these hidden notes are important in different contexts. At least I would need the invis property.

```
function anno_out(s, t, f) {
       //this condition is what I propose
       if (s.invis==true){
             return;
        }

	if (s.istart == undefined)
		return
	var	type = s.type,
		h = s.ymx - s.ymn + 4,
		wl = s.wl || 2,
		wr = s.wr || 2

	if (s.grace)
		type = GRACE

	f(t || anno_type[type], s.istart, s.iend,
		s.x - wl - 2, staff_tb[s.st].y + s.ymn + h - 2,
		wl + wr + 4, h, s);
}

```

**Comment by moinejf** [comment]
The problem was due to an old mechanism about the alternate note heads.
I did some cleanup in the code and this should be fixed by the commit cfd07dbbd4829615dda25ea4905a22f2e342afb5.
Thanks.

**Comment by spheras** [comment]
Hi, 

Thank you for the fix. I have cloned and compiled the code to check it, and I've seen something extrange, but I don't know if it is related with the fix or just another different problem from the master branch. When I try to load the ABC reported some errors appears at the console:

`platform-browser.es5.js:2914 Error: <path> attribute d: Expected number, "m60.90 NaN\nl0.68 NaNv-3…".
platform-browser.es5.js:2914 Error: <path> attribute d: Expected number, "m61.24 NaN\nl0.00 NaNv-3…".`

It is due a **NaN** number, I've tried to follow the thread and it seems a problem when calculating the beam.a and beam.b properties.

`function calculate_beam(bm, s1)`

the bm.a and bm.b are both NaN after calling this function. As I said, maybe this is a different ticket, I don't know. This only seems to happen in the previously ABC code reported.

Hope it helps.

**Comment by moinejf** [comment]
Thanks for the report.
The NaN problem was raised by the previous commit. This should be fixed by the commit 4900517ea1ed15fd9a55e0538aac9041ec1ef04f.
Also, music split on beam should be fixed by the commit 6ff768da9f091e553ec6736ef7e914df0c6e0340.
